src/components/calcite-dropdown/calcite-dropdown.tsx

```diff
@@ -95,18 +130,31 @@ export class CalciteDropdown {
     }
   }
 
-  componentWillLoad() {
-    // get initially selected items
-    this.updateSelectedItems();
+  disconnectedCallback(): void {
+    this.destroyPopper();
   }
 
-  render() {
+  render(): VNode {
     const { maxScrollerHeight } = this;
     const dir = getElementDir(this.el);
     return (
-      <Host dir={dir} tabIndex={this.disabled ? -1 : null}>
-        <slot name="dropdown-trigger" aria-haspopup="true" aria-expanded={this.active.toString()} />
+      <Host dir={dir}>
         <div
+          class="calcite-dropdown-trigger-container"
+          ref={(el) => (this.referenceEl = el)}
```

Comment:
Nitpick: This doesn't have a big impact because it's just assigning the element, but this anonymous function will get called on each render. Moving it to a class method will guarantee it gets called once.

Reply:
This is likely to fail the travis build too.

Reply:
Also, is there any reason it is preferable to use `refs` instead of just doing `this.el.shadowRoot.querySelector(".calcite-dropdown-trigger-container")`?

Reply:
Yes, the ref will return the node when it's created. If you use querySelector you are doing a query through the DOM after creation which is slower.

Reply:
If we're querying through shadowRoot we should replace those to use ref wherever possible.

Reply:
> This is likely to fail the travis build too.

I'm not sure why it would fail. Could you elaborate on that?

Reply:
Gotcha. Weird, linting errors should halt commits. ⛔